commit 37d779e66bc0358d91dfed5f7ed9deeb0996cc41 (refs/changes/60/360/3)
Author: Todd Whiteman <todd.whiteman@joyent.com>
Date:   2016-08-26T17:00:03-07:00 (3 years, 1 month ago)
    
    DOCKER-920 Docker build shouldn't need to use chown after copying files
    Reviewed by: Joshua M. Clulow <jmc@joyent.com>
    Reviewed by: Josh Wilsdon <jwilsdon@joyent.com>
    Reviewed by: Trent Mick <trentm@gmail.com>

diff --git a/bin/docker-build.js b/bin/docker-build.js
index 6b7555e..cdacea1 100644
--- a/bin/docker-build.js
+++ b/bin/docker-build.js
@@ -273,6 +273,17 @@ function runBuild(opts) {
 
     log.info('%s - started for zone %s', commandType, opts.uuid);
 
+    // Before we start, we want all files created by docker build/commit to use
+    // 'root:root' permissions. Note that the cn-agent (parent) process defaults
+    // to using 'root:staff' permissions.
+    // We do this so that we can avoid having to 'chown' files that get added by
+    // the ADD/COPY instructions (that way we don't need to keep track of what
+    // was copied and we avoid the performance of calling chown).
+    // Note that this should not be needed if we change the tar extraction to
+    // run inside of the zone - DOCKER-872.
+    process.setuid(0);
+    process.setgid(0);
+
     sendMessage('Starting ' + commandType, {socket: socket, log: log});
 
     if (commandType === 'commit') {
@@ -745,7 +756,7 @@ function handleExtractTarfileEvent(builder, event) {
     var tarfile = event.tarfile;
 
     // Ensure the holding (parent) directory exists.
-    builder.mkdirpChown(extractDir, function (err) {
+    mkdirp(extractDir, function (err) {
         if (err) {
             callback(err);
             return;
